rosbag2_py/test/test_transport.py

```diff
@@ -40,3 +46,50 @@ def test_options_qos_conversion():
     record_options = rosbag2_py.RecordOptions()
     record_options.topic_qos_profile_overrides = simple_overrides
     assert record_options.topic_qos_profile_overrides == simple_overrides
+
+
+def test_record_cancel(tmp_path):
+    bag_path = str(tmp_path / 'test_record_cancel')
+    storage_options, converter_options = get_rosbag_options(bag_path)
+
+    recorder = rosbag2_py.Recorder()
+
+    record_options = rosbag2_py.RecordOptions()
+    record_options.all = True
+    record_options.is_discovery_disabled = False
+    record_options.rmw_serialization_format = ""
+    record_options.topic_polling_interval = datetime.timedelta(milliseconds=100)
+
+    record_options.compression_mode = 'file'
+    record_options.compression_queue_size = 1
+    record_options.compression_threads = 0
```

Comment:
I'd remove the compression options, the defaults are reasonable, and `file` compression adds extra complexity we don't need in this test

```suggestion
```